### PR TITLE
[CMake-improve] Build Android with sync enabled on CI

### PR DESCRIPTION
**Pull request by fealebenpae**

### CODE REVIEW

File: tests/util/event_loop.cpp [hunk 1]
```diff
@@ -204,6 +205,69 @@ void EventLoop::Impl::perform(std::function<void()> f)
...
+static void release_main_looper()
+{
+    ALooper_release(s_main_looper);
+}
+
+bool EventLoop::has_implementation() { return true; }
+
+std::unique_ptr<EventLoop::Impl> EventLoop::Impl::main()
+{
+    return std::unique_ptr<Impl>(new Impl(s_main_looper));
```
Comment (bdash) [inline tests/util/event_loop.cpp]: May as well use `std::make_unique` here.

File: tests/util/event_loop.cpp [hunk 2]
```diff
@@ -59,10 +53,17 @@ struct EventLoop::Impl {
...
     Impl(util::CFPtr<CFRunLoopRef> loop) : m_loop(std::move(loop)) { }
 
     util::CFPtr<CFRunLoopRef> m_loop;
+#elif REALM_USE_ALOOPER
+    Impl(ALooper* looper);
+    void perform_work();
+
+    std::vector<std::function<void()>> m_pending_work;
+    std::mutex m_mutex;
+    std::shared_ptr<util::EventLoopSignal<std::function<void()>>> m_signal;
```
Comment (bdash) [inline tests/util/event_loop.cpp]: Why does this need shared ownership? It looks to be owned exclusively by this `Impl` type.

Comment (fealebenpae) [reply tests/util/event_loop.cpp]: The implementation of the Android `EventLoopSignal` relies on it being owned by a `std::shared_ptr`, as far as I can tell.

Comment (bdash) [reply tests/util/event_loop.cpp]: Ah, that explains it!

File: tests/util/event_loop.cpp (hunk 1 above)
Comment (fealebenpae) [reply tests/util/event_loop.cpp]: I tried that, but the `Impl` constructor is private.

**Pull request merged**